carbon-design-system/carbon-addons-iot-react
# fix(suite-header): do not allow tabbing to closed app switcher panel

### Merging into `master` to make it into 8.7

Closes #3217 

**Summary**

- The ApplicationSwitcher buttons had no awareness if the panel was open or closed, so I pass the `isExpanded` prop to the AppSwitcher component in order to change the tabindex as necessary.

**Change List (commits, features, bugs, etc)**

- passes `isExpanded` to panel elements that are react components (not HTML elements)
- change tabIndex on appswitcher buttons based on isExpanded
- add tests to confirm

**Acceptance Test (how to verify the PR)**

- Go to suite header story
- open dev tools
- inspect the application switcher 9-dot button
- add the `bx--header-panel--expanded` class to the div beside it (has `data-testid="action-btn__panel"`)
- tab through the page and confirm the focus does not drop into the application switcher menu from the application switcher 9-dot button
-  click the 9-dot button to set the state and let the component know it is _actually_ open
- tab through again and confirm focus jumps into the appswitcher panel this time

**Regression Test (how to make sure this PR doesn't break old functionality)**

- make sure app switcher still works as expected with keyboard navigation



**Things to look for during review**
